packages/extended-services-vscode-extension/src/configurations/Configuration.ts

```diff
@@ -19,81 +19,48 @@
+export const extendedServicesURLID = "extendedServices.extendedServicesURL";
 
 export class Configuration {
   readonly enableAutoRun: boolean;
-  readonly connectionHeartbeatIntervalinSecs: number;
+  readonly connectionHeartbeatIntervalInSecs: number;
   readonly extendedServicesURL: URL;
 
-  constructor(enableAutoRun: boolean, connectionHeartbeatIntervalinSecs: number, extendedServicesURL: URL) {
+  constructor(enableAutoRun: boolean, connectionHeartbeatIntervalInSecs: number, extendedServicesURL: URL) {
     this.enableAutoRun = enableAutoRun;
-    this.connectionHeartbeatIntervalinSecs = connectionHeartbeatIntervalinSecs;
+    this.connectionHeartbeatIntervalInSecs = connectionHeartbeatIntervalInSecs;
     this.extendedServicesURL = extendedServicesURL;
   }
 }
 
-function fetchEnableAutoRun(): boolean {
-  const enableAutoRun = vscode.workspace.getConfiguration().get<boolean>(enableAutoRunID);
-  if (!enableAutoRun) {
-    throw new Error("Enable Auto Run configuration not found");
-  }
-  return enableAutoRun;
-}
-
-function fetchConnectionHeartbeatIntervalinSecs(): number {
-  const connectionHeartbeatIntervalinSecs = vscode.workspace
-    .getConfiguration()
-    .get<number>(connectionHeartbeatIntervalinSecsID);
-  if (!connectionHeartbeatIntervalinSecs) {
-    throw new Error("Connection Heartbeat Interval configuration not found");
-  }
-
-  return connectionHeartbeatIntervalinSecs;
-}
-
 function fetchExtendedServicesURL(): URL {
-  const extendedServicesURL = vscode.workspace.getConfiguration().get<string>(extendedServicesURLID);
-  if (!extendedServicesURL) {
-    throw new Error("URL configuration not found");
-  }
-
+  const defaultExtendedServicesURL = `http://${process.env.WEBPACK_REPLACE__extendedServicesUrlHost}:${process.env.WEBPACK_REPLACE__extendedServicesUrlPort}`;
+  const extendedServicesURL = getConfigurationPropertyValue<string>(extendedServicesURLID, defaultExtendedServicesURL);
   try {
     return new URL(extendedServicesURL);
   } catch (error) {
-    throw new Error("Invalid service URL:" + error.message);
+    throw new Error(`URL configuration ${extendedServicesURL} is invalid: ${error.message}`);
   }
 }
 
-export function fetchConfiguration(): Configuration {
-  let errorMessages: string[] = [];
-  let enableAutoRun: any;
-  let connectionHeartbeatIntervalinSecs: any;
-  let extendedServicesURL: any;
-
-  try {
-    enableAutoRun = fetchEnableAutoRun();
-  } catch (error) {
-    errorMessages.push(error.message);
-  }
-
-  try {
-    connectionHeartbeatIntervalinSecs = fetchConnectionHeartbeatIntervalinSecs();
-  } catch (error) {
-    errorMessages.push(error.message);
+const getConfigurationPropertyValue = <T>(property: string, defaultValue: T): T => {
+  const value: T | null = vscode.workspace.getConfiguration().get(property) as T;
+  if (value == null) {
+    console.warn(`Property: ${property} is missing, using the default: ${defaultValue}`);
+    value == defaultValue;
   }
+  return value;
+};
```

Comment:
@yesamer I just have one comment related to this method, here is a suggestion of improvement. Also, I'm fixing two things here, the first double equals (`==`) to triple equals, and the second double equals should be an attribution.

```suggestion
// Place on top of the file or on another file
// I don't think this is a suitable name, it's just to exemplify
type Property = string | number | boolean

const getConfigurationPropertyValue = <T extends Property | null>(property: string, defaultValue: T): T => {
  const value = vscode.workspace.getConfiguration().get(property) as T;
  if (value === null) {
    console.warn(`Property: ${property} is missing, using the default: ${defaultValue}`);
    value = defaultValue;
  }
  return value;
};
```

Reply:
@ljmotta Thank you, I committed your suggested change, I'll update the comment and the property name